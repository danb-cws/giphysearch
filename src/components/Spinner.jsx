import React from "react";

const Spinner = props => {
  if (props.isLoaded) {
    return <p className="Spinner">data loaded, spinner off..</p>;
  } else {
    return <p className="Spinner">spin spin spinner on...</p>;
  }
}
export default Spinner;