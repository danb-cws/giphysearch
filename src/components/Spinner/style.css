.Spinner {
  background: rgba(0, 0, 0, 0.9) var(--spinner-b64-url) no-repeat center center;
  width: 100%;
  height: 100%;
}
