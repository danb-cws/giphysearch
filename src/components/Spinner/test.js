import React from "react";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";
import Spinner from "./";

describe("Spinner tests", () => {
  it("Shows spinner if showSpinner is true", () => {
    const el = shallow(<Spinner showSpinner={true} />);
    expect(el.contains(<div className="Spinner" />)).toBe(true);
  });

  it("Renders nothing if showSpinner is false", () => {
    const el = shallow(<Spinner showSpinner={false} />);
    expect(el.find(Spinner).exists()).toEqual(false);
  });

  it("Renders and matches snapshot", () => {
    const component = renderer.create(<Spinner showSpinner={true} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
