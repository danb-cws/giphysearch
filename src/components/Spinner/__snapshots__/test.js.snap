// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner tests Renders and matches snapshot 1`] = `
<div
  className="Spinner"
/>
`;
